Tribler/community/market/conversion.py

```diff
@@ -0,0 +1,233 @@
+                                 self._encode_offer, self._decode_offer)
+        self.define_meta_message(chr(2), community.get_meta_message(u"bid"),
+                                 self._encode_offer, self._decode_offer)
+        self.define_meta_message(chr(3), community.get_meta_message(u"cancel-order"),
+                                 self._encode_cancel_order, self._decode_cancel_order)
+        self.define_meta_message(chr(4), community.get_meta_message(u"offer-sync"),
+                                 self._encode_offer_sync, self._decode_offer_sync)
+        self.define_meta_message(chr(5), community.get_meta_message(u"proposed-trade"),
+                                 self._encode_proposed_trade, self._decode_proposed_trade)
+        self.define_meta_message(chr(6), community.get_meta_message(u"declined-trade"),
+                                 self._encode_declined_trade, self._decode_declined_trade)
+        self.define_meta_message(chr(7), community.get_meta_message(u"counter-trade"),
+                                 self._encode_proposed_trade, self._decode_proposed_trade)
+        self.define_meta_message(chr(8), community.get_meta_message(u"start-transaction"),
+                                 self._encode_start_transaction, self._decode_start_transaction)
+        self.define_meta_message(chr(9), community.get_meta_message(u"wallet-info"),
+                                 self._encode_wallet_info, self._decode_wallet_info)
+        self.define_meta_message(chr(10), community.get_meta_message(u"payment"),
+                                 self._encode_payment, self._decode_payment)
+        self.define_meta_message(chr(11), community.get_meta_message(u"end-transaction"),
+                                 self._encode_transaction, self._decode_transaction)
+
+    def _encode_introduction_request(self, message):
+        data = BinaryConversion._encode_introduction_request(self, message)
+
+        if message.payload.orders_bloom_filter:
+            data.extend((pack('!BH', message.payload.orders_bloom_filter.functions,
+                              message.payload.orders_bloom_filter.size), message.payload.orders_bloom_filter.prefix,
+                         message.payload.orders_bloom_filter.bytes))
+        return data
+
+    def _decode_introduction_request(self, placeholder, offset, data):
+        offset, payload = BinaryConversion._decode_introduction_request(self, placeholder, offset, data)
+
+        if len(data) > offset:
+            if len(data) < offset + 5:
+                raise DropPacket("Insufficient packet size")
+
+            functions, size = unpack_from('!BH', data, offset)
+            offset += 3
+
+            prefix = data[offset]
+            offset += 1
+
+            if functions <= 0 or size <= 0 or size % 8 != 0:
+                raise DropPacket("Invalid bloom filter")
+
+            length = int(ceil(size / 8))
```

Comment:
Why not just `size/8`? You already checked if `size %8 == 0` so `ceil` is not needed, furthermore size was already unpacked as `int` through `H`.